util/jsonrpc-types/src/blockchain.rs

```diff
@@ -514,6 +514,43 @@ impl From<Transaction> for packed::Transaction {
     }
 }
 
+/// The JSON view of a transaction as well as its status.
+#[derive(Serialize, Deserialize, PartialEq, Eq, Hash, Debug)]
+pub struct TransactionWithStatusResponse {
+    /// The transaction.
+    pub transaction: Option<ResponseFormat<TransactionView>>,
+    /// The Transaction status.
+    pub tx_status: TxStatus,
+}
+
+impl TransactionWithStatusResponse {
+    /// Create a new `TransactionWithStatusResponse` with the json format TransactionView.
+    pub fn derive_json_tx(t: TransactionWithStatus) -> Self {
+        Self {
+            transaction: t.transaction.map(ResponseFormat::json),
+            tx_status: t.tx_status,
+        }
+    }
+
+    /// Create a new `TransactionWithStatusResponse` with the hex format TransactionView.
+    pub fn derive_hex_tx(t: TransactionWithStatus) -> Self {
+        Self {
+            transaction: t.transaction.map(|v| {
+                let tx = packed::Transaction::from(v.inner).into_view();
+                ResponseFormat::hex(
+                    packed::TransactionView::new_builder()
+                        .data(tx.data())
+                        .witness_hash(tx.witness_hash())
+                        .hash(tx.hash())
+                        .build()
+                        .as_bytes(),
+                )
+            }),
+            tx_status: t.tx_status,
+        }
+    }
+}
+
 /// The JSON view of a transaction as well as its status.
 #[derive(Clone, Serialize, Deserialize, PartialEq, Eq, Hash, Debug)]
 pub struct TransactionWithStatus {
```

Comment:
Here is also a legacy issue, in fact, the tx-pool should not use the JSON type directly.
Here we can move `TransactionWithStatus` to util/types/src/core/tx_pool.rs.
```
pub transaction: Option<TransactionView>,
```
This field is changed to Option<core::TransactionView>.
This avoids the problem of repeated serialization.